oars_gb_pkg/nodes/sensing/wind_filter.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+import rospy
+import numpy as np
+from collections import deque
+from std_msgs.msg import UInt8
+from geometry_msgs.msg import Pose2D
+
+'''
+This code subscribes to /sensors/airmar/wind/rel and /sensors/airmar/wind/true
+and stores on a rolling basis the most recent 10 values into a deque (or a rolling
+que). The code converts that deque into a numpy array and performs an
+average function on the array. The resulting values are a rolling average, and
+they are published to /weather/wind/rel and /weather/wind/true
+'''
+
+# Create deques to store last 10 values
+wind_rel_deque = {'x': deque(maxlen=10), 'theta': deque(maxlen=10)}
+wind_true_deque = {'x': deque(maxlen=10), 'theta': deque(maxlen=10)}
+
+# Create publishers for relative and true wind speed topics
+pub_rel = rospy.Publisher('/weather/wind/rel', Pose2D, queue_size=10)
+pub_true = rospy.Publisher('/weather/wind/true', Pose2D, queue_size=10)
+
+def airmar_subscriber():
+    rospy.init_node("airmar_wind_filter", anonymous=True) # Initialize nodes
+
+    # Subscribe to airmar node's topics
+    rospy.Subscriber("/sensors/airmar/wind/rel", Pose2D, rel_filter)
+    rospy.Subscriber("/sensors/airmar/wind/true", Pose2D, true_filter)
+    rospy.Subscriber("/control/airmar_filter/deque_size", UInt8, deque_resizer)
+
+    rospy.spin()
+
+
+def deque_resizer(d_size):
+    print('We\'re in')
+    d_size = d_size.data
+    if wind_rel_deque['x'].maxlen != d_size:
+        wind_rel_deque['x'] = deque(wind_rel_deque['x'], maxlen=d_size)
+        wind_rel_deque['theta'] = deque(wind_rel_deque['theta'], maxlen=d_size)
+        wind_true_deque['x'] = deque(wind_true_deque['x'], maxlen=d_size)
+        wind_true_deque['theta'] = deque(wind_true_deque['theta'], maxlen=d_size)
+
+def rel_filter(rel_wind):
+    # Append read value to deque
+    wind_rel_deque['x'].append(rel_wind.x)
+    wind_rel_deque['theta'].append(rel_wind.theta)
+
+    len_check = wind_rel_deque['x'].maxlen
+    # Only run if the deque has already stored 10 values (passed initialization period)
```

Comment:
Only if this is true, do what? It'd be nice if I didn't have to read all the code in the block below to figure out what is happening in the `if` statement (e.g. "Only publish a wind message if we've finished initializing").